### PR TITLE
Worker Table for the worker manager UI

### PR DESCRIPTION
**Pull request by owlishDeveloper**

### CODE REVIEW

File: ui/src/components/WMWorkersTable/index.jsx [hunk 1]
```diff
@@ -0,0 +1,257 @@
...
+import DateDistance from '../DateDistance';
+import sort from '../../utils/sort';
+import { WMWorkers } from '../../utils/prop-types';
+
+@withRouter
+export default class WorkerManagerWorkersTable extends Component {
+  static propTypes = {
+    searchTerm: String,
+    workers: WMWorkers.isRequired,
+  };
```
Comment (helfi92) [inline ui/src/components/WMWorkersTable/index.jsx]: For non required `propTypes`, it would be nice to have `defaultProps` defined.

```jsx
static propTypes = {
 // ...
};

static defaultProps = {
  searchTerm: '',
};
```

File: ui/src/components/WMWorkersTable/index.jsx [hunk 2]
```diff
@@ -0,0 +1,257 @@
...
+          <TableCell>
+            <Typography>n/a</Typography>
+          </TableCell>
+        )}
+
+        <TableCell>
+          <TableCellListItem
+            button
+            component={Link}
+            to={`${path}/worker-types/${workerType}/workers/${workerGroup}/${workerId}/recent-errors`}>
```
Comment (helfi92) [inline ui/src/components/WMWorkersTable/index.jsx]: Clicking turns the url into `http://localhost:5080/worker-manager/worker-types/:workerType/worker-types/banana/workers/%F0%9F%A6%80/rust-awesomness/recent-errors`. `worker-types` seems to be duplicated and there is `:workerType`.

File: ui/src/components/WMWorkersTable/index.jsx [hunk 3]
```diff
@@ -0,0 +1,257 @@
...
+            />
+            <AlertIcon size={iconSize} />
+          </TableCellListItem>
+        </TableCell>
+
+        <TableCell>
+          <TableCellListItem
+            button
+            component={Link}
+            to={`${path}/worker-types/${workerType}/workers/${workerGroup}/${workerId}/resources`}>
```
Comment (helfi92) [inline ui/src/components/WMWorkersTable/index.jsx]: I see a similar behavior to the previous comment re: "Click to see errors".

File: ui/src/utils/prop-types.js [hunk 4]
```diff
@@ -180,6 +180,31 @@ export const workers = shape({
   edges: arrayOf(worker),
 });
 
+export const WMWorkers = arrayOf(
+  shape({
```
Comment (helfi92) [inline ui/src/utils/prop-types.js]: In this file, we could define what a `WMWorker` looks like and then in the component file we could say that it is an array of `WMWorker`. In other words, instead of specifying `arrayOf` in this file, we would do it in `WMWorkersTable/index.jsx `.

File: services/web-server/src/graphql/WorkerManager.graphql [hunk 5]
```diff
@@ -16,6 +16,25 @@ type WorkerManagerWorkerTypeSummary {
...
+
+    latestTaskRun: TaskRun
+
+    quarantineUntil: DateTime
+
+    recentErrors: Int
+
+    providerId: String!
+    workerType: String!
+}
```
Comment (helfi92) [inline services/web-server/src/graphql/WorkerManager.graphql]: Do we know which properties will require an extra request and which won't? If `providerId` and `workerType` don't trigger an extra request, they could be brought up below `workerAge` without the newline.

File: services/web-server/src/loaders/workerManager.js [hunk 6]
```diff
@@ -12,6 +12,16 @@ module.exports = ({ workerManager }) => {
...
 
+  // const WMWorkers = new DataLoader(queries =>
+  //   Promise.all(
+  //     queries.map(async (workerType, isQuarantined, filter) => {
+  //       const workers = await workerManager.getWorkers(provisionerId, workerType, isQuarantined);
+  //
+  //       return filter ? sift(filter, workers) : workers;
+  //     })
+  //   )
+  // );
```
Comment (helfi92) [inline services/web-server/src/loaders/workerManager.js]: Do you want me to look at the validity of these comments?

Comment (owlishDeveloper) [reply services/web-server/src/loaders/workerManager.js]: not yet - these are just sketches at the moment. I plan to get them finished when I switch to API, and when I'll clean up this from the mock data

File: ui/src/components/WMWorkersTable/index.jsx (hunk 2 above)
Comment (owlishDeveloper) [reply ui/src/components/WMWorkersTable/index.jsx]: Oh, that link is supposed to direct you to the list of the recent tasks for the worker. I'll remove the link altogether for now

File: ui/src/components/WMWorkersTable/index.jsx (hunk 3 above)
Comment (owlishDeveloper) [reply ui/src/components/WMWorkersTable/index.jsx]: yeah, same thing. I'll remove these links altogether for now.

File: ui/src/utils/prop-types.js (hunk 4 above)
Comment (owlishDeveloper) [reply ui/src/utils/prop-types.js]: what are the advantages to this method?

File: services/web-server/src/graphql/WorkerManager.graphql (hunk 5 above)
Comment (owlishDeveloper) [reply services/web-server/src/graphql/WorkerManager.graphql]: Not yet (at least, I'm not concentrating on that yet). I organized this with spaces to have the following structure:

`new fields from worker manager`
`data from queue`
`I'm not even sure about this one`
`two fields that "roll over" from the previous table`

It's more for my convenience than for anything. If you like, I can add this to the clean up ticket to remove all the white spaces later

Comment (helfi92) [reply services/web-server/src/graphql/WorkerManager.graphql]: Adding it to the clean up ticket works for me.

File: ui/src/utils/prop-types.js (hunk 4 above)
Comment (helfi92) [reply ui/src/utils/prop-types.js]: Advantage: If a future view will be responsible to render a single worker manager worker, then it would be able to grab the `WMWorker` structure from here.
